src/pages/test-report/test-report.analytics.effects.ts

```diff
@@ -39,10 +37,73 @@ export class TestReportAnalyticsEffects {
 
   @Effect()
   testReportViewDidEnter$ = this.actions$.pipe(
-    ofType(TEST_REPORT_VIEW_DID_ENTER),
-    switchMap((action: TestReportViewDidEnter) => {
+    ofType(testReportActions.TEST_REPORT_VIEW_DID_ENTER),
+    switchMap((action: testReportActions.TestReportViewDidEnter) => {
       this.analytics.setCurrentPage(AnalyticsScreenNames.TEST);
-      return of({});
+      return of(new AnalyticRecorded());
+    }),
+  );
+
+  @Effect()
+  toggleRemoveFaultMode$ = this.actions$.pipe(
+    ofType(testReportActions.TOGGLE_REMOVE_FAULT_MODE),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getTests),
+        map(isTestReportPracticeTest),
+      ),
+    ),
+    concatMap(([action, isTestReportPracticeTest]: [testReportActions.ToggleRemoveFaultMode, boolean]) => {
+      if (!isTestReportPracticeTest) {
+        this.analytics.logEvent(
+          AnalyticsEventCategories.TEST_REPORT,
+          AnalyticsEvents.SELECT_REMOVE_MODE,
+        );
+        return of(new AnalyticRecorded());
+      }
+      return of(new AnalyticNotRecorded());
+    }),
+  );
+
+  @Effect()
+  toggleSeriousFaultMode$ = this.actions$.pipe(
+    ofType(testReportActions.TOGGLE_SERIOUS_FAULT_MODE),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getTests),
+        map(isTestReportPracticeTest),
+      ),
+    ),
+    concatMap(([action, isTestReportPracticeTest]: [testReportActions.ToggleSeriousFaultMode, boolean]) => {
+      if (!isTestReportPracticeTest) {
+        this.analytics.logEvent(
+          AnalyticsEventCategories.TEST_REPORT,
+          AnalyticsEvents.SELECT_SERIOUS_MODE,
+        );
+        return of(new AnalyticRecorded());
+      }
+      return of(new AnalyticNotRecorded());
+    }),
+  );
+
+  @Effect()
+  toggleDangerousFaultMode$ = this.actions$.pipe(
+    ofType(testReportActions.TOGGLE_DANGEROUS_FAULT_MODE),
+    withLatestFrom(
+      this.store$.pipe(
+        select(getTests),
+        map(isTestReportPracticeTest),
```

Comment:
If your trying to avoid practice mode this selector on catches the Test Report one not the End To End . You need to use this selector instead - isPracticeMode